bioio_ome_zarr/writers/ome_zarr_writer_v3.py

```diff
@@ -0,0 +1,541 @@
+    """
+    OMEZarrWriterV3 is a fully compliant OME-Zarr v0.5.0 writer built
+    on Zarr v3 stores. Supports 2 ≤ N ≤ 5 dimensions (e.g. YX, ZYX,
+    TYX, CZYX, or TCZYX).
+    """
+
+    def __init__(
+        self,
+        store: Union[str, zarr.storage.StoreLike],
+        shape: Tuple[int, ...],
+        dtype: Union[np.dtype, str],
+        axes_names: Optional[List[str]] = None,
+        axes_types: Optional[List[str]] = None,
+        axes_units: Optional[List[Optional[str]]] = None,
+        axes_scale: Optional[List[float]] = None,
+        scale_factors: Optional[Tuple[int, ...]] = None,
+        num_levels: Optional[int] = None,
+        chunks: Union[str, Tuple[int, ...], List[Tuple[int, ...]]] = "auto",
+        shards: Optional[Union[Tuple[int, ...], List[Tuple[int, ...]]]] = None,
+        compressor: Optional[BloscCodec] = None,
+        image_name: str = "Image",
+        channels: Optional[List[Channel]] = None,
+        rdefs: Optional[dict] = None,
+        creator_info: Optional[dict] = None,
+        multiscale_scale: Optional[List[float]] = None,
+    ):
+        """
+        Initialize writer and build axes + channel metadata automatically.
+
+        Parameters
+        ----------
+        store : Union[str, zarr.storage.StoreLike]
+            Path or Zarr store-like object for the output group.
+        shape : Tuple[int, ...]
+            Base image shape (e.g. (Y, X), (Z, Y, X), (T, C, Z, Y, X)).
+        dtype : Union[np.dtype, str]
+            NumPy dtype of the image data (e.g. "uint8").
+        axes_names : Optional[List[str]]
+            Names of each axis, len = len(shape). Defaults to last N of
+            ["t", "c", "z", "y", "x"].
+        axes_types : Optional[List[str]]
+            Types of each axis (e.g. ["time", "channel", "space"...]).
+        axes_units : Optional[List[Optional[str]]]
+            Physical units for each axis (e.g. ["ms", None, "µm"]).
+        axes_scale : Optional[List[float]]
+            Physical scale per axis at base resolution.
+        scale_factors : Optional[Tuple[int, ...]]
+            Integer downsampling factors per axis (e.g. (1, 1, 2, 2)).
+        num_levels : Optional[int]
+            Number of pyramid levels to generate. If None, compute until
+            no further reduction.
+        chunks : Union[str, Tuple[int, ...], List[Tuple[int, ...]]]
+            Chunk sizes: "auto" or a tuple or list per level.
+        shards : Optional[Union[Tuple[int, ...], List[Tuple[int, ...]]]]
+            Shard factors per level. None for no sharding.
+        compressor : Optional[BloscCodec]
+            Zarr compressor to use (default: Blosc Zstd).
+        image_name : str
+            Name for the image in multiscales metadata.
+        channels : Optional[List[Channel]]
+            OMERO-style channel metadata objects.
+        rdefs : Optional[dict]
+            OMERO rendering settings (passed under "omero" → "rdefs").
+        creator_info : Optional[dict]
```

Comment:
this is cool, it would be great to specify even more provenance goodness inside of here!

Reply:
issue created